src/OrchardCore.Modules/OrchardCore.Forms/Views/Items/ButtonPart.cshtml

```diff
@@ -0,0 +1,10 @@
+@using OrchardCore.Forms.Models
+@model ShapeViewModel<ButtonPart>
+@{
+    var formElementPart = Model.Value.ContentItem.As<FormElementPart>();
+    var formInputElementPart = Model.Value.ContentItem.As<FormInputElementPart>();
+    var elementId = formElementPart.Id;
+    var elementName = formInputElementPart.Name;
+    var id = !string.IsNullOrWhiteSpace(elementId) ? elementId : !string.IsNullOrWhiteSpace(elementName) ? Html.GenerateIdFromName(elementName) : default(string);
```

Comment:
Validation/sanitazation should prevent the whitespace in the form, then here you can just check `NullOrEmpty`

Reply:
Ok